src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -698,9 +699,50 @@ export default class QuestionnaireForm extends Component {
     return c;
   }
 
+  // use this method
+
+  addAuthorToResponse(qr, practionerRef)
+  {
+    // url is a string
+    function addAuthor(item, practionerRef)
+    {
+      item["extension"] =
+    [
+        {
+            "url": "http://hl7.org/fhir/StructureDefinition/questionnaireresponse-author",
+            "valueReference": 
+            {
+                "reference": practionerRef
+            }
+        }
+    ]
+
+      console.log(item);
+      console.log(practionerRef);
+      }
+
+      qr.item.map(item=> 
+        {
+            item.item.map(item =>
```

Comment:
The questionnaire items are recursive.  You can put items inside other items, and though we only ever go 2 deep in our example questionnaires, you could nest items as much as you want.  If you want to search through all the items in the questionnaire response, you have to do it recursively.  You can check the `prepopulate()` function for an example, since it also looks through every item in the questionnaire.